Source/WebCore/dom/Element.cpp

```diff
@@ -3220,11 +3259,13 @@ ExceptionOr<ShadowRoot&> Element::attachShadow(const ShadowRootInit& init)
         init.serializable ? ShadowRoot::Serializable::Yes : ShadowRoot::Serializable::No,
         isPrecustomizedOrDefinedCustomElement() ? ShadowRoot::AvailableToElementInternals::Yes : ShadowRoot::AvailableToElementInternals::No,
         WTFMove(registry), init.registry ? ShadowRoot::ScopedCustomElementRegistry::Yes : ShadowRoot::ScopedCustomElementRegistry::No);
+    if (document().settings().shadowRootReferenceTargetEnabled())
```

Comment:
Is this settings check superfluous?

Reply:
I think so, since `setReferenceTarget` rightfully does a settings check.